### PR TITLE
networkmanager: Better behavior with unavailable devices

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

**Comment by mvollmer** [comment]
testBond is failing.

**Comment by mvollmer** [comment]
> testBond is failing.

Fixed.

**Comment by stefwalter** [comment]
What does this do for the user? Does it need a screenshot?

**Comment by stefwalter** [comment]
What does this do for the user? Does it need a screenshot?

**Comment by mvollmer** [comment]
> What does this do for the user? Does it need a screenshot?

It avoids erratic behavior in the UI with devices that NM labels as "unavailable".  We didn't explicitly consider this state before.

Cockpit would never find the connection settings for a unavailable device.  Changing the settings would complete without an error, but not have any effect in the UI because the new settings weren't found.  Instead, each settings change would create a new file in /etc/sysconfig/network-scripts/.

Also, we know that a unavailable device can not be activated.

**Comment by mvollmer** [comment]
> Does it need a screenshot?

I don't think so.

File: pkg/networkmanager/interfaces.js [hunk 1]
```diff
@@ -2421,7 +2425,7 @@ PageNetworkInterface.prototype = {
             $('#network-interface-mac').text(mac);
         }
 
-        this.device_onoff.onoff("disabled", !iface);
+        this.device_onoff.onoff("disabled", !!(!iface || (dev && dev.State == 20)));
```
Comment (dperpeet) [inline pkg/networkmanager/interfaces.js]: where does this `20` come from?

Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: from the NetworkManager D-Bus API documentation.

Comment (dperpeet) [inline pkg/networkmanager/interfaces.js]: and I think the boolean expression could use a comment
if you add one here, I can insert during merge

**Comment by dperpeet** [comment]
Is this behavior we should add in a test?

**Comment by mvollmer** [comment]
> Is this behavior we should add in a test?

Ideally, yes, but I don't know how to make a unavailable device with NetworkManager...

File: pkg/networkmanager/interfaces.js (hunk 1 above)
Comment (mvollmer) [inline pkg/networkmanager/interfaces.js]: Comment: Disable the On/Off button for interfaces that we don't know about at all, and for devices that NM declares to be unavailable.  Neither can be activated.

**Comment by stefwalter** [comment]
We need @mvollmer to bring this to green ... so unmarking priority.

**Comment by mvollmer** [comment]
> We need @mvollmer to bring this to green ... so unmarking priority.

I don't think I can do much more than retrying tests atm.

**Comment by mvollmer** [comment]
This failure:
```
connection.id: property is missing
```
is very interesting, but has very likely not been introduced by this PR.  It's probably from #5870. I try to figure it out...

**Pull request closed without merging**